scripts/west_commands/runners/ezflashcli.py

```diff
@@ -39,13 +39,22 @@ def do_create(cls, cfg, args):
         return EzFlashCliBinaryRunner(cfg, tool=args.tool, sn=args.sn,
                                       erase=args.erase)
 
+    def get_load_offset(self):
+        if self.build_conf.getboolean('CONFIG_MCUBOOT'):
+            return 0
+        if self.build_conf.getboolean('CONFIG_HAS_FLASH_LOAD_OFFSET'):
+            return self.build_conf['CONFIG_FLASH_LOAD_OFFSET']
+        return 0
+
     def program_bin(self):
+        load_offset = self.get_load_offset()
+
         if self.erase:
             self.logger.info("Erasing flash...")
-            self.check_call([self.tool] + self.sn_arg + ["erase_flash"])
-
-        self.logger.info(f"Flashing {self.bin_}...")
-        self.check_call([self.tool] + self.sn_arg + ["image_flash", self.bin_])
+        if load_offset > 0:
+            self.check_call([self.tool] + self.sn_arg + ["write_flash", f'0x{load_offset:x}', self.bin_])
```

Comment:
Can this not flash a hex file instead? I don't know the layout of this chip but if there is a large gap in the memory layout, a hex file is going to be far smaller than a bin file, plus a hex file is transferrable to anywhere and can be programmed, a bin file is basically useless unless you have the zephyr project files.

Reply:
ezFlashCLI can only write bin files. It's all written to single location without gaps so file size is not a problem.